### PR TITLE
added two new embedding model's encoding

### PR DESCRIPTION
**Pull request by Praneet460**
**Problem**
Library doesn't support two new embedding model's encoding mapper
- `text-embedding-3-small`
- `text-embedding-3-large`

`tiktoken.encoding_for_model("text-embedding-3-small") ` raises a **KeyError**

Screenshot 2024-01-27 at 1 08 05 AM

**Solution**
Added Encoding mapper for 2 new embedding models. The source of mapping is taken from [here](https://github.com/openai/openai-cookbook/blob/main/examples/How_to_count_tokens_with_tiktoken.ipynb)

### CODE REVIEW

**Comment by hoonlight** [comment]
@hauntsaninja Hi, can you check this PR?

**Comment by stevieflyer** [comment]
Really looking forward to this merge

**Comment by zafercavdar** [comment]
Is there any public documentation which clarifies that the new embedding models have cl100k_base encoding?

File: tiktoken/model.py [hunk 1]
```diff
@@ -26,6 +26,8 @@
     "babbage-002": "cl100k_base",
     # embeddings
     "text-embedding-ada-002": "cl100k_base",
+    "text-embedding-3-small": "cl100k_base",
+    "text-embedding-3-large": "cl100k_base",
```
Comment (zafercavdar) [inline tiktoken/model.py:30]: Is there any public documentation which clarifies that the new embedding models have cl100k_base encoding?

Comment (zafercavdar) [reply tiktoken/model.py:30]: It seems that they're clarified here: https://platform.openai.com/docs/guides/embeddings/embedding-models

Comment (Praneet460) [reply tiktoken/model.py:30]: @zafercavdar they are mentioning it in their cookbook
Attaching a source https://github.com/openai/openai-cookbook/blob/main/examples/How_to_count_tokens_with_tiktoken.ipynb
Screenshot 2024-02-01 at 5 11 33 AM

**Review by emsi — APPROVED** [review]

File: tiktoken/model.py (hunk 1 above)
Comment (Praneet460) [reply tiktoken/model.py:30]: @zafercavdar Can you approve the PR

**Review by tisserasuneth — APPROVED** [review]
looks good!

**Comment by chrispy-snps** [comment]
Thanks for adding these!

**Review by kawada711 — APPROVED** [review]
+

**Comment by usamasaleem1** [comment]
Can we merge this to main so I can start using the new models!

**Comment by Vaibhav2001** [comment]
> Can we merge this to main so I can start using the new models!

+1

**Comment by tartom7997** [comment]
+1

**Comment by jnance314** [comment]
+1

**Comment by emsi** [comment]
For the meantime you can just:
`pip install -U git+https://github.com/Praneet460/tiktoken@Add-New-Embedding-Models`

**Comment by will-mako-ai** [comment]
+1

**Review by hauntsaninja — APPROVED** [review]
Thank you!

**Comment by hauntsaninja** [comment]
This has been released in tiktoken 0.6

**Pull request merged**